src/adaptations/device-layer/include/Weave/DeviceLayer/OpenThread/GenericThreadStackManagerImpl_OpenThread.ipp

```diff
@@ -182,11 +191,27 @@ WEAVE_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetThreadEnabl
     Impl()->LockThreadStack();
 
     bool isEnabled = (otThreadGetDeviceRole(mOTInst) != OT_DEVICE_ROLE_DISABLED);
+    bool isIp6Enabled = otIp6IsEnabled(mOTInst);
+
+    if (val && !isIp6Enabled)
+    {
+        otErr = otIp6SetEnabled(mOTInst, val);
+        VerifyOrExit(otErr == OT_ERROR_NONE, );
```

Comment:
SuccessOrExit(otErr) instead ?

Reply:
I would absolutely agree if this was a `WEAVE_ERROR`.  For `OT_ERROR` I went with this version.